avocado/core/multiplexer.py

```diff
@@ -377,28 +377,67 @@ def iteritems(self):
                 yield (leaf.environment_origin[key].path, key, value)
 
 
+def _report_already_parsed(self, *args, **kwargs):
+    """
+    Raises exception describing that `self.data` alteration is restricted
+    """
+    raise RuntimeError("Altering of `self.data` is restricted. %s %s"
```

Comment:
Right, I remember hitting this before.  It's unfortunate that we have to make code workarounds to support pickling classes...  *Ideally* either pickle would be more robust, or we'd keep data separate from code.